### PR TITLE
Test jenkins-infra/github-reusable-workflows#31

### PR DESCRIPTION
**Pull request by lemeurherve**
This PR is a test for https://github.com/jenkins-infra/github-reusable-workflows/pull/31.

Trying to set java-version to 17 first with a check in cd GHA logs on merge, then I'll try with default value.

### Testing done

```[tasklist]
### Submitter checklist
- [ ] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [ ] Ensure that the pull request title represents the desired changelog entry
- [ ] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue
```

### CODE REVIEW

**Comment by lemeurherve** [comment]
Not sure to understand the goal of this original code:
```
  check_run:
    types:
      - completed
```

It seems to run the cd GHA in loop with it cf https://github.com/jenkinsci/jenkins-infra-test-plugin/actions

image

Is it expected?

**Comment by timja** [comment]
Its not a loop its just triggered when each check run is completed.

The build triggers 14 checks apparently:

image

GitHub doesn't have a built-in way to only trigger when a check of x name (`Jenkins`) is triggered.

File: .github/workflows/cd.yml [hunk 1]
```diff
@@ -9,7 +9,9 @@ on:
 
 jobs:
   maven-cd:
-    uses: jenkins-infra/github-reusable-workflows/.github/workflows/maven-cd.yml@v1
+    uses: lemeurherve/github-reusable-workflows/.github/workflows/maven-cd.yml@v1
+    with:
+      java-version: 17
```
Comment (timja) [inline .github/workflows/cd.yml:14]: should probably change the plugin to java 17 as well to do a full test?

Comment (lemeurherve) [reply .github/workflows/cd.yml:14]: Is something like that sufficient?

https://github.com/jenkinsci/flyway-api-plugin/pull/9/files#diff-9c5fb3d1b7e3b0f54bc5c4182965c4fe1f9023d449017cece3005d3f90e8e4d8R36-R37

Comment (timja) [reply .github/workflows/cd.yml:14]: yes

**Comment by lemeurherve** [comment]
https://github.com/jenkins-infra/github-reusable-workflows/pull/31 has been merged, closing.

**Pull request closed without merging**